src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -860,7 +860,7 @@ public function morphTo($name = null, $type = null, $id = null)
         // If the type value is null it is probably safe to assume we're eager loading
         // the relationship. When that is the case we will pass in a dummy query as
         // there are multiple types in the morph and we can't use single queries.
-        if (is_null($class = $this->$type)) {
+        if (is_null($class = $this->$type) || $class === '') {
```

Comment:
Maybe this change isn't correct then, and our tests should be failing here too but the tests are just incomplete?